### PR TITLE
Correct mishandling of encoder parameters

### PR DESCRIPTION
**Pull request by mytianer**
Hello, when I use this library, I found a lot of problems, many formats cannot be directly converted successfully using the following code:
try AudioConverter.convert(file.url, to: URL(fileURLWithPath: "xxxxx/xxxx.xxx"))

Others
1. Opus and speex encode work well, but there is a problem with decoding playback
2. There seems to be a problem when ogg and oga encode
3. The use of custom encoder parameters for mp3 has no effect

I'm not proficient with c and c++ code, I don't know if I'm using the wrong method or there is an error in itself

### CODE REVIEW

File: Encoders/SFBCoreAudioEncoder.mm [hunk 1]
```diff
@@ -438,7 +438,7 @@ - (BOOL)openReturningError:(NSError **)error
 		}
 
 		if(audioConverter) {
-			for(NSNumber *key in audioConverterPropertySettings) {
+			for(NSNumber *key in audioConverterPropertySettings.allKeys) {
```
Comment (sbooth) [inline Encoders/SFBCoreAudioEncoder.mm]: This change is not necessary. Fast enumeration for NSDictionary iterates over the keys: https://developer.apple.com/documentation/foundation/nsdictionary?language=objc

File: Metadata/SFBAudioMetadata.m [hunk 2]
```diff
@@ -601,7 +601,7 @@ - (void)copyAttachedPicturesFrom:(SFBAudioMetadata *)metadata
 - (NSArray *)attachedPicturesOfType:(SFBAttachedPictureType)type
 {
 	NSMutableArray *pictures = [NSMutableArray array];
-	for(SFBAttachedPicture *picture in _pictures) {
+    for(SFBAttachedPicture *picture in _pictures.copy) {
```
Comment (sbooth) [inline Metadata/SFBAudioMetadata.m]: I'm unclear on the reasoning behind this change. At least in the context of this PR I don't think it belongs?

**Comment by sbooth** [comment]
Thanks for this PR. You've certainly found a few bugs and most of your changes are correct. Please see my other comments on a few of the specifics.

Regarding the other issues you raise, could you open a separate issue for each one for easier tracking?

File: Metadata/SFBAudioMetadata.m (hunk 2 above)
Comment (mytianer) [reply Metadata/SFBAudioMetadata.m]: sorry, i submit the wrong place. It must be in the removeAttachedPicturesOfType mehtod

File: Encoders/SFBCoreAudioEncoder.mm (hunk 1 above)
Comment (mytianer) [reply Encoders/SFBCoreAudioEncoder.mm]: thank you, i get it.

**Comment by mytianer** [comment]
I have already modified 2 files

File: Metadata/SFBAudioMetadata.m [hunk 3]
```diff
@@ -620,7 +620,7 @@ - (void)removeAttachedPicture:(SFBAttachedPicture *)picture
 
 - (void)removeAttachedPicturesOfType:(SFBAttachedPictureType)type
 {
-	for(SFBAttachedPicture *picture in _pictures) {
+	for(SFBAttachedPicture *picture in _pictures.copy) {
```
Comment (sbooth) [inline Metadata/SFBAudioMetadata.m]: Could you explain the reasoning behind this change? I don't think it's necessary.

Comment (mytianer) [reply Metadata/SFBAudioMetadata.m]: Deleting elements while looping through an array causes a crash

Comment (sbooth) [reply Metadata/SFBAudioMetadata.m]: What about something like:

```objc
	NSSet *picturesToRemove = [_pictures objectsPassingTest:^BOOL(SFBAttachedPicture *obj, BOOL *stop) {
#pragma unused(stop)
		return obj.pictureType == type;
	}];
	[_pictures minusSet:picturesToRemove];
```

Comment (mytianer) [reply Metadata/SFBAudioMetadata.m]: It's pretty good

**Review by sbooth — APPROVED** [review]

**Pull request closed without merging**